debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresHeartbeat.java

```diff
@@ -0,0 +1,64 @@
+package io.debezium.connector.postgresql;
+
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.connector.postgresql.connection.PostgresConnection;
+import io.debezium.function.BlockingConsumer;
+import io.debezium.heartbeat.HeartbeatImpl;
+
+/**
+ * Postgres specific implementaiton of the heartbeat feature that allows for a PG query to be executed with every heartbeat.
+ */
+public class PostgresHeartbeat extends HeartbeatImpl {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostgresHeartbeat.class);
+
+    public static final String HEARTBEAT_ACTION_ENABLED_PROPERTY_NAME = "heartbeat.action.enabled";
+    public static final String HEARTBEAT_ACTION_QUERY_PROPERTY_NAME = "heartbeat.action.query";
+
+    public static final Field HEARTBEAT_ACTION_ENABLED = Field.create(HEARTBEAT_ACTION_ENABLED_PROPERTY_NAME)
```

Comment:
Do we need a separate option for it? Would not be enought to have presence/nonpresence of HEARTBEAT_ACTION_QUERY ?

Reply:
I was conflicted about this. I see this feature as something people might want to enable / disable for testing etc. I guess they could just set the query to be empty to disable but I like the more explicit configuration. Then again I can see the argument for fewer configuration options.

Reply:
Yes, presence of the query should be enough to enable this feature.